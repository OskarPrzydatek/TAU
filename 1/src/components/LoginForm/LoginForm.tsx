import React from 'react';
import SubmitButton from '../Button/SubmitButton';
import { Input } from '../Input/Input';

type LoginFormProps = {
  onLoginSubmit: (email: string, password: string) => void;
};

export default function LoginForm({ onLoginSubmit }: LoginFormProps) {
  const emailLoginFormRef = React.useRef<HTMLInputElement>(null);
  const passwordLoginFormRef = React.useRef<HTMLInputElement>(null);

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    const email = emailLoginFormRef.current?.value;
    const password = passwordLoginFormRef.current?.value;

    if (email && password) {
      onLoginSubmit(email, password);
    }
  };

  return (
    <form onSubmit={onSubmit}>
      <h2>Login</h2>
      <Input
        label="Email"
        placeholder="Put email here :D"
        ref={emailLoginFormRef}
        inputTestId="login-form-email-input"
      />
      <Input
        label="Password"
        type="password"
        placeholder="Put email here :D"
        ref={passwordLoginFormRef}
        inputTestId="login-form-password-input"
      />
      <SubmitButton label="Login" dataTestId="login-form-submitt-button" />
    </form>
  );
}
