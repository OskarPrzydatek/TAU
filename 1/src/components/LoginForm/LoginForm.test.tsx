import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';

const mockOnLoginForm = jest.fn();

describe('LoginForm', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<LoginForm onLoginSubmit={mockOnLoginForm} />);
    expect(tree).toMatchSnapshot();
  });

  test('ensure form data handlig works correctly', () => {
    render(<LoginForm onLoginSubmit={mockOnLoginForm} />);
    fireEvent.click(screen.getByTestId('login-form-submitt-button'));
    expect(mockOnLoginForm).not.toHaveBeenCalled();
    fireEvent.change(screen.getByTestId('login-form-email-input'), {
      target: { value: 'name@mail.com' },
    });
    fireEvent.change(screen.getByTestId('login-form-password-input'), {
      target: { value: 'qwerty123' },
    });
    fireEvent.click(screen.getByTestId('login-form-submitt-button'));
    expect(mockOnLoginForm).toHaveBeenCalledWith('name@mail.com', 'qwerty123');
  });
});

export {};
