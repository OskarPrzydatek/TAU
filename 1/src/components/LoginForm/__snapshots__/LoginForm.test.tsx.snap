// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <label>
    <span>
      Email
    </span>
    <input
      data-testid="login-form-email-input"
      placeholder="Put email here :D"
      type="text"
    />
  </label>
  <label>
    <span>
      Password
    </span>
    <input
      data-testid="login-form-password-input"
      placeholder="Put email here :D"
      type="password"
    />
  </label>
  <button
    data-testid="login-form-submitt-button"
    type="submit"
  >
    Login
  </button>
</form>
`;
